infra/experimental/sanitizers/ExecSan/README.md

```diff
@@ -1,18 +1,23 @@
 # Shell Injection Detection with `ptrace`
 
-We use `ptrace` to instrument system calls made by the target program to detect if our `/tmp/tripwire` command in `vuln.dict` was injected into the shell of the testing target program and executed by the program to produce of a `/tmp/injected` file.
-Our instrumentation verifies the existence of `/tmp/injected` after every `execve` or each process spawned via `clone`, which proves the existence of shell injection vulnerabilities.
+We use `ptrace` to instrument system calls made by the target program to detect
+if our `/tmp/tripwire` command in `vuln.dict` was injected into the shell of
+the testing target program. This works by
+
+- Checking if `execve` is called with `/tmp/tripwire`.
+- TODO: Checking if we managed to invoke a shell (e.g. /bin/sh) and cause a
+  syntax error.
 
 ## Quick test
 
 ### Cleanup
-Note this will delete /tmp/tripwire and /tmp/injected if they exist
+Note this will delete /tmp/tripwire if it exists.
 ```shell
 make clean
 ```
 
 ### Run test
-Note this will overwrite /tmp/tripwire and /tmp/injected if they exist
+Note this will overwrite /tmp/tripwire and if they exist.
```

Comment:
"and" what?

Reply:
I think the end of this sentence should be "/tmp/tripwire if it exists."

Reply:
flxed